skfda/representation/basis/_fdatabasis.py

```diff
@@ -912,6 +912,11 @@ def __rtruediv__(
         """Right division for FDataBasis object."""
         return NotImplemented
 
+    def __neg__(self: T) -> T:
+        """Negation of FData object."""
+        return self._copy_op(other=None, coefficients=-self.coefficients)
```

Comment:
`self._copy_op` is only really needed for binary operations. It is used to consider the case where a `FData` with `n_samples == 1` and other with `n_samples > 1` are operated together. In that case you want to copy the `FData` with `n_samples > 1`, as the result will have the same number of samples. In this case, as there is only one operand, you can directly use `self.copy` instead.